pulp_smash/tests/rpm/api_v2/utils.py

```diff
@@ -42,3 +46,82 @@ def sync_repo(server_config, href):
         urljoin(href, 'actions/sync/'),
         {'override_config': {}}
     )
+
+
+def xml_handler(_, response):
+    """API response handler for fetching XML generated by yum distributor.
+
+    Checks status code and returns an ElementTree for the response.
+
+    Note that the entire response XML is loaded and parsed before returning,
+    so this may be unsafe for use with large XML files.
+
+    If the response URL ends in .gz, the XML will be automatically
+    decompressed.
+
+    Currently ignores Content-Type and Content-Encoding headers from server
+    due to https://pulp.plan.io/issues/1781 .
+    """
+    response.raise_for_status()
+
+    # It would be better if this were based on server response headers and
+    # not guessed on filename.
+    # See https://pulp.plan.io/issues/1781
+    is_gzip = True if response.request.url.endswith('.gz') else False
+
+    if is_gzip and 'decompress' in dir(gzip):
+        # python 3
+        text = gzip.decompress(response.content)  # pylint: disable=no-member
+        text = str(text, 'utf-8')
+        return ElementTree.fromstring(text)
+    elif is_gzip:
+        # python 2
+        compressed = io.BytesIO(response.content)
+        decompressed = gzip.GzipFile(fileobj=compressed)
+        return ElementTree.parse(decompressed)
```

Comment:
I was wondering about that when reviewing this yesterday. Thanks for the clarification.